guides/common/modules/proc_enabling-the-project-client-name-repository-for-rhel-9-and-rhel-8.adoc

```diff
@@ -0,0 +1,55 @@
+[id="enabling-the-project-client-name-repository-rhel-9-and-rhel-8"]
+= Enabling the {project-client-name} repository for {RHEL} 9 and {RHEL} 8
+
+To use the CLI instead of the {ProjectWebUI}, see the procedure relevant for your {RHEL} version:
+
+* xref:CLI_Enabling_the_Client_Repository_rhel_9_{context}[]
+* xref:CLI_Enabling_the_Client_Repository_rhel_8_{context}[]
+
+ifeval::["{mode}" == "disconnected"]
+.Prerequisites
+* Ensure that you import all content ISO images that you require into {ProjectServer}.
+endif::[]
+
+.Procedure
+. In the {ProjectWebUI}, navigate to *Content* > *Red Hat Repositories*.
+. In the Available Repositories pane, enable the *Recommended Repositories* to get the list of repositories.
+. Click *{Team} {project-client-name} for RHEL 9 x86_64 (RPMs)* or *{Team} {project-client-name} for RHEL 8 x86_64 (RPMs)* to expand the repository set.
+. For the *x86_64* architecture, click the *+* icon to enable the repository.
++
+If the *{project-client-name}* items are not visible, it may be because they are not included in the Red{nbsp}Hat Subscription Manifest obtained from the Customer Portal.
+To correct that, log in to the Customer Portal, add these repositories, download the Red{nbsp}Hat Subscription Manifest and import it into {Project}.
+For more information, see {ContentManagementDocURL}Managing_Red_Hat_Subscriptions_content-management[Managing Red Hat Subscriptions] in _{ContentManagementDocTitle}_.
++
+Enable the {project-client-name} repository for every supported major version of {RHEL} running on your hosts.
+After enabling a Red Hat repository, a Product for this repository is automatically created.
+
+[id="CLI_Enabling_the_Client_Repository_rhel_9_{context}"]
+.CLI procedure for {RHEL} 9
+* Enable the {project-client-name} repository:
++
+[options="nowrap" subs="+quotes,attributes"]
+----
+# hammer repository-set enable \
+--basearch="x86_64" \
+--name "{Team} {project-client-name} for RHEL 9 x86_64 (RPMs)" \
+--organization "_My_Organization_" \
+--product "{RHEL} for x86_64"
+----
+
+[id="CLI_Enabling_the_Client_Repository_rhel_8_{context}"]
+.CLI procedure for {RHEL} 8
+* Enable the {project-client-name} repository:
++
+[options="nowrap" subs="+quotes,attributes"]
+----
+# hammer repository-set enable \
+--basearch="x86_64" \
+--name "{Team} {project-client-name} for RHEL 8 x86_64 (RPMs)" \
+--organization "_My_Organization_" \
+--product "{RHEL} for x86_64"
+----
```

Comment:
I came up with the way to modularize this even further, but I understand if you won't accept this suggestion, since the same cannot really be done for RHEL 7 and 6 without making it more messy.
```diff
diff --git a/guides/common/modules/proc_enabling-the-project-client-name-repository-for-rhel-7-and-rhel-6.adoc b/guides/common/modules/proc_enabling-the-project-client-name-repository-for-rhel-7-and-rhel-6.adoc
index 0ebf403ed1..f545ff4e6a 100644
--- a/guides/common/modules/proc_enabling-the-project-client-name-repository-for-rhel-7-and-rhel-6.adoc
+++ b/guides/common/modules/proc_enabling-the-project-client-name-repository-for-rhel-7-and-rhel-6.adoc
@@ -56,4 +56,4 @@ After enabling a Red Hat repository, a Product for this repository is automatica
 
 .Additional resources
 
-* For details about the `hammer repository-set enable` command, enter `hammer repository-set enable --help`.
\ No newline at end of file
+* For details about the `hammer repository-set enable` command, enter `hammer repository-set enable --help`.
diff --git a/guides/common/modules/proc_enabling-the-project-client-name-repository-for-rhel-9-and-rhel-8.adoc b/guides/common/modules/proc_enabling-the-project-client-name-repository-for-rhel-9-and-rhel-8.adoc
index 89561581d1..83ed9fa69d 100644
--- a/guides/common/modules/proc_enabling-the-project-client-name-repository-for-rhel-9-and-rhel-8.adoc
+++ b/guides/common/modules/proc_enabling-the-project-client-name-repository-for-rhel-9-and-rhel-8.adoc
@@ -24,31 +24,13 @@ For more information, see {ContentManagementDocURL}Managing_Red_Hat_Subscription
 Enable the {project-client-name} repository for every supported major version of {RHEL} running on your hosts.
 After enabling a Red Hat repository, a Product for this repository is automatically created.
 
-[id="CLI_Enabling_the_Client_Repository_rhel_9_{context}"]
-.CLI procedure for {RHEL} 9
-* Enable the {project-client-name} repository:
-+
-[options="nowrap" subs="+quotes,attributes"]
-----
-# hammer repository-set enable \
---basearch="x86_64" \
---name "{Team} {project-client-name} for RHEL 9 x86_64 (RPMs)" \
---organization "_My_Organization_" \
---product "{RHEL} for x86_64"
-----
-
-[id="CLI_Enabling_the_Client_Repository_rhel_8_{context}"]
-.CLI procedure for {RHEL} 8
-* Enable the {project-client-name} repository:
-+
-[options="nowrap" subs="+quotes,attributes"]
-----
-# hammer repository-set enable \
---basearch="x86_64" \
---name "{Team} {project-client-name} for RHEL 8 x86_64 (RPMs)" \
---organization "_My_Organization_" \
---product "{RHEL} for x86_64"
-----
+:repo-rhel-major-version: 9
+include::snip_enabling-the-project-client-name-repository-cli.adoc[]
+
+:repo-rhel-major-version: 8
+include::snip_enabling-the-project-client-name-repository-cli.adoc[]
+
+:!repo-rhel-major-version:
 
 .Additional resources
 
diff --git a/guides/common/modules/proc_synchronizing-the-project-client-name-repository-for-rhel-9-and-rhel-8.adoc b/guides/common/modules/proc_synchronizing-the-project-client-name-repository-for-rhel-9-and-rhel-8.adoc
index 295c554dec..7b9826f94b 100644
--- a/guides/common/modules/proc_synchronizing-the-project-client-name-repository-for-rhel-9-and-rhel-8.adoc
+++ b/guides/common/modules/proc_synchronizing-the-project-client-name-repository-for-rhel-9-and-rhel-8.adoc
@@ -12,29 +12,13 @@ To use the CLI instead of the {ProjectWebUI}, see the procedure relevant for you
 . Select *{Team} {project-client-name} for RHEL 9 x86_64 RPMs* or *{Team} {project-client-name} for RHEL 8 x86_64 RPMs*.
 . Click *Synchronize Now*.
 
-[id="CLI_Synchronizing_the_Client_Repository_rhel_9_{context}"]
-.CLI procedure for {RHEL} 9
-* Synchronize your {project-client-name} repository:
-+
-[options="nowrap" subs="+quotes,attributes"]
-----
-# hammer repository synchronize \
---name "{Team} {project-client-name} for RHEL 9 x86_64 RPMs" \
---organization "_My_Organization_" \
---product "{RHEL} for x86_64"
-----
-
-[id="CLI_Synchronizing_the_Client_Repository_rhel_8_{context}"]
-.CLI procedure for {RHEL} 8
-* Synchronize your {project-client-name} repository:
-+
-[options="nowrap" subs="+quotes,attributes"]
-----
-# hammer repository synchronize \
---name "{Team} {project-client-name} for RHEL 8 x86_64 RPMs" \
---organization "_My_Organization_" \
---product "{RHEL} for x86_64"
-----
+:repo-rhel-major-version: 9
+include::snip_synchronizing-the-project-client-name-repository-cli.adoc[]
+
+:repo-rhel-major-version: 8
+include::snip_synchronizing-the-project-client-name-repository-cli.adoc[]
+
+:!repo-rhel-major-version:
 
 .Additional resources
 
diff --git a/guides/common/modules/snip_enabling-the-project-client-name-repository-cli.adoc b/guides/common/modules/snip_enabling-the-project-client-name-repository-cli.adoc
new file mode 100644
index 0000000000..df4075146b
--- /dev/null
+++ b/guides/common/modules/snip_enabling-the-project-client-name-repository-cli.adoc
@@ -0,0 +1,12 @@
+[id="CLI_Enabling_the_Client_Repository_rhel_{repo-rhel-major-version}_{context}"]
+.CLI procedure for {RHEL} {repo-rhel-major-version}
+* Enable the {project-client-name} repository:
++
+[options="nowrap" subs="+quotes,attributes"]
+----
+# hammer repository-set enable \
+--basearch="x86_64" \
+--name "{Team} {project-client-name} for RHEL {repo-rhel-major-version} x86_64 (RPMs)" \
+--organization "_My_Organization_" \
+--product "{RHEL} for x86_64"
+----
diff --git a/guides/common/modules/snip_synchronizing-the-project-client-name-repository-cli.adoc b/guides/common/modules/snip_synchronizing-the-project-client-name-repository-cli.adoc
new file mode 100644
index 0000000000..ec07bda49d
--- /dev/null
+++ b/guides/common/modules/snip_synchronizing-the-project-client-name-repository-cli.adoc
@@ -0,0 +1,11 @@
+[id="CLI_Synchronizing_the_Client_Repository_rhel_{repo-rhel-major-version}_{context}"]
+.CLI procedure for {RHEL} {repo-rhel-major-version}
+* Synchronize your {project-client-name} repository:
++
+[options="nowrap" subs="+quotes,attributes"]
+----
+# hammer repository synchronize \
+--name "{Team} {project-client-name} for RHEL {repo-rhel-major-version} x86_64 RPMs" \
+--organization "_My_Organization_" \
+--product "{RHEL} for x86_64"
+----
```

Reply:
I'd prefer not to go this far. Mostly because I prefer to be really conservative with snippets. (They can make things messy and IMO it's best to use them only when a piece of content is repeated many times and in different parts of the docs set.) So I'd stick with the basic modularization in this case if you're okay with it.

But in any case, thanks for sending this patch, it was a path worth investigating :)